src/utils.js

```diff
@@ -44,71 +44,125 @@ function dbExists (dbName) {
   })
 }
 
+/**
+ *
+ * @param {string} key - The key (/prefix/key)
+ * @param {string} nonce - The nonce is used to ensure privacy and avoid de-anonymization (hex string)
+ */
+const hashKey = async (key, nonce) => {
+  if (key === '/' || key === '') return '/'
+  const prefixes = key.split('/').filter(prefix => prefix !== '')
```

Comment:
ok

Reply:
what if key is `//mykey//` ?

Reply:
Why filter all empty strings ? is a key like '/abc//def' valid ?